spec/requests/cart_items_request_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'CartItems', type: :request do
+  describe 'PUT /cart_items/:id' do
+    let!(:unauthorized_user) { create :user }
+    let!(:product) { create :product }
+    let!(:item) { create :cart_item, :with_cart }
+
+    before do
+      sign_in item.user
+    end
+
+    it 'updates cart_items quantity to 12 if user is the cart owner' do
+      put "/cart_items/#{item.id}", params: { cart_item: { quantity: 12 } }
+      follow_redirect!
+      expect(response.body).to include('Item updated successfully')
+      expect(item.reload.quantity).to eq(12)
+    end
+
+    it 'does not update item if the user is not the cart owner' do
+      sign_out item.user
```

Comment:
I think it will be better if you use context here.
For specs when user is authorized and when is not authorized to do it

Reply:
Good idea! Now it is much more readable and I could extract some methods to `before` blocks.